odl/space/pspace.py

```diff
@@ -1064,6 +1123,158 @@ def ufuncs(self):
         """
         return ProductSpaceUfuncs(self)
 
+    @property
+    def real(self):
+        """Real part of the element.
+
+        Examples
+        --------
+        >>> space = odl.ProductSpace(odl.cn(3), odl.cn(2))
+        >>> x = space.element([[1 + 1j, 2, 3 - 3j], [-1+2j, -2-3j]])
+        >>> x_real = x.real
+        >>> real_space = odl.ProductSpace(odl.rn(3), odl.rn(2))
+        >>> expected_result = real_space.element([[1, 2, 3], [-1, -2]])
+        >>> x_real == expected_result
+        True
+        """
+        real_part = [part.real for part in self.parts]
+        return self.space.real_space.element(real_part)
+
+    @real.setter
+    def real(self, newreal):
+        """Setter for the real part.
+
+        This method is invoked by ``x.real = other``.
```

Comment:
Related to the comment below, should we mention what happens with the imaginary part of an element which one tries to set as real part, i.e., if someone does something like `x.real = 1 + 1j`?

Reply:
In my opinion it should throw an error. Is there a reason it does not? Have you tested what numpy does, since we want to do the exact same.

Reply:
```python
>>> arr = np.zeros(5)
>>> arr.imag = 3
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: array does not have imaginary part to set
>>> arr.real = 2j
__main__:1: ComplexWarning: Casting complex values to real discards the imaginary part
```

this is what we expect basically

Reply:
I'd even say that the exact behavior depends on the backend, and that we shouldn't add logic that interferes with that. So IMO we pass on whatever we get and let the error or warning bubble up.